### PR TITLE
[PREVIEW] SL-1564 sort problems by descending date of creation within severity

### PR DESCRIPTION
**Pull request by zielinskikamil**

### CODE REVIEW

File: src/app/problems/containers/problems/problems-page.component.spec.ts [hunk 1]
```diff
@@ -100,4 +91,39 @@ describe('ProblemsPageComponent', () => {
...
+                criticalProblem,
+                urgentProblem,
+                warningProblem,
+                undefinedSeverityProblem
+            ])
+        });
+    });
 });
+
+function problemGenerator(createdAt: moment.Moment, severity = 'Warning'): Problem {
```
Comment (filipczyks) [inline src/app/problems/containers/problems/problems-page.component.spec.ts]: can we name it generateProblem, createProblem etc? function names should be verbs

File: src/app/problems/containers/problems/problems-page.component.html [hunk 2]
```diff
@@ -1,3 +1,3 @@
 <app-problems-table
-        [problems]="problems$ | async">
+        [problems]="problems">
```
Comment (kpakur) [inline src/app/problems/containers/problems/problems-page.component.html]: why not async?

File: src/app/problems/components/problems-table/problems-table.component.ts [hunk 3]
```diff
@@ -9,23 +9,25 @@ import { Moment } from 'moment';
...
   displayedColumns = ['severity', 'createdAt', 'message', 'references description'];
-  dataSource;
+  dataSource: MatTableDataSource<Problem>;
 
-  @Input() problems: Problem[];
+  private _problems: Problem[];
+  @Input() set problems(problems: Problem[]) {
+    this._problems = problems;
+    this.dataSource = new MatTableDataSource(Object.values(this.problems));
+    this.dataSource.paginator = this.paginator;
```
Comment (kpakur) [inline src/app/problems/components/problems-table/problems-table.component.ts]: 👍

File: src/app/problems/components/problems-table/problems-table.component.ts [hunk 4]
```diff
@@ -9,23 +9,25 @@ import { Moment } from 'moment';
...
 
   @ViewChild(MatPaginator) paginator: MatPaginator;
 
   constructor() {}
 
-  ngOnInit() {}
-
-  ngOnChanges() {
-    this.dataSource = new MatTableDataSource(Object.values(this.problems));
-    this.dataSource.paginator = this.paginator;
```
Comment (kpakur) [inline src/app/problems/components/problems-table/problems-table.component.ts]: 👍

File: src/app/problems/containers/problems/problems-page.component.ts [hunk 5]
```diff
@@ -1,41 +1,18 @@
...
-            map(this.sortByCreatedAtDescending));
-    }
+    constructor(private readonly problemsService: ProblemsService) { }
 
     ngOnInit() {
-        this.store.dispatch(new fromProblemsPartsActions.Get())
+        this.problemsService.getAll().subscribe(problems => this.problems = problems);
     }
-
-    sortByCreatedAtDescending(problems: Problem[]): Problem[] {
```
Comment (kpakur) [inline src/app/problems/containers/problems/problems-page.component.ts]: 👍

File: src/app/problems/components/problems-table/problems-table.component.ts (hunk 3 above)
Comment (radoslawlandowskikainos) [reply src/app/problems/components/problems-table/problems-table.component.ts]: 👍

File: src/app/problems/components/problems-table/problems-table.component.ts (hunk 4 above)
Comment (radoslawlandowskikainos) [reply src/app/problems/components/problems-table/problems-table.component.ts]: 👍

**Pull request merged**